third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -1133,9 +1133,14 @@ impl<'env> Generator<'env> {
                     // Type checker should have complained already
                     self.internal_error(id, "inconsistent tuple arity")
                 } else {
-                    // Map this to point-wise assignment
-                    for (pat, exp) in pats.iter().zip(args.iter()) {
-                        self.gen_assign(id, pat, exp, next_scope)
+                    // Save each tuple arg into a temporary.
```

Comment:
No concern that probably in 98% of the cases new temps are not needed? Should we add a quick check like `lhs.vars() * rhs.vars() != 0`?

Reply:
I think variable coalescing + dead store elimination would catch most of these cases, so it is likely not an issue. However, it is good to not generate too many temps - if it is possible without adding too much complexity.

So, now, **only** when we have n-tuple with n>1 and overlapping LHS and RHS variables (like you suggested), we add the temps.

There are at least two more cases where we can avoid generating temps even when we have overlapping variables, but I felt adding all those checks would increase complexity (and they would be eliminated by optimizations anyway).